### ISSUE TITLE
Show duplicate app on android

### ISSUE THREAD

**Issue by AndonMitev** [auth0/react-native-auth0#571]
I ran into an issue where after properly logging in on Android (device or emulator), I am presented with duplicate copies of my app and prompted to select which of the two copies I would like to use to open the link. One seems to work, and the other seems to return an error.

running `adb shell dumpsys package r` will output this:

```
dev.ooo.opn.wallet:
        975ba9a dev.ooo.opn.wallet/.MainActivity
        a64e666 dev.ooo.opn.wallet/com.auth0.react.RedirectActivity
```

Intent filters:

```
 <intent-filter>
        <action android:name="android.intent.action.MAIN"/>
        <category android:name="android.intent.category.LAUNCHER"/>
      </intent-filter>
      <intent-filter>
        <action android:name="android.intent.action.VIEW"/>
        <category android:name="android.intent.category.DEFAULT"/>
        <category android:name="android.intent.category.BROWSABLE"/>
         <data android:scheme="defiwallet"/>
        <data android:scheme="dev.ooo.opn.wallet"/>
        <data android:scheme="exp+defi-wallet"/>
      </intent-filter>
     <intent-filter android:autoVerify="true" data-generated="true">
        <action android:name="android.intent.action.VIEW"/>
        <data android:scheme="https" android:host="1b1.art" android:pathPrefix="/claim"/>
        <category android:name="android.intent.category.BROWSABLE"/>
        <category android:name="android.intent.category.DEFAULT"/>
      </intent-filter>
      <intent-filter data-generated="true">
        <action android:name="android.intent.action.VIEW"/>
         <data android:scheme="dev.ooo.opn.wallet"/>
        <data android:scheme="exp+defi-wallet"/>
        <category android:name="android.intent.category.BROWSABLE"/>
        <category android:name="android.intent.category.DEFAULT"/>
      </intent-filter>
```

Callback url: `dev.ooo.opn.wallet://{MY_DOMAIN}/android/dev.ooo.opn.wallet/callback`

Actual:

https://user-images.githubusercontent.com/32274987/207081814-d9d635f8-ce7d-4cf8-8585-79d479d2be98.png

Expected:

 To have only one or automatically open the app both works as expected

Env:

```
"react-native-auth0": "2.15.0",
"expo": "~46.0.17",
"react-native": "0.69.6"
```

**Comment by poovamraj** [comment]
Hi @AndonMitev, the issue is that you are using the same custom scheme to open your `MainActivity` as the one used in the redirection. Hope this helps. I will close this issue as it was confirmed by you as well.

**Comment by erisvaldojunior** [comment]
> I ran into an issue where after properly logging in on Android (device or emulator), I am presented with duplicate copies of my app and prompted to select which of the two copies I would like to use to open the link. One seems to work, and the other seems to return an error.
> 
> 
> 
> running `adb shell dumpsys package r` will output this:
> 
> 
> 
> ```
> 
> dev.ooo.opn.wallet:
> 
>         975ba9a dev.ooo.opn.wallet/.MainActivity
> 
>         a64e666 dev.ooo.opn.wallet/com.auth0.react.RedirectActivity
> 
> ```
> 
> 
> 
> Intent filters:
> 
> 
> 
> ```
> 
>  <intent-filter>
> 
>         <action android:name="android.intent.action.MAIN"/>
> 
>         <category android:name="android.intent.category.LAUNCHER"/>
> 
>       </intent-filter>
> 
>       <intent-filter>
> 
>         <action android:name="android.intent.action.VIEW"/>
> 
>         <category android:name="android.intent.category.DEFAULT"/>
> 
>         <category android:name="android.intent.category.BROWSABLE"/>
> 
>          <data android:scheme="defiwallet"/>
> 
>         <data android:scheme="dev.ooo.opn.wallet"/>
> 
>         <data android:scheme="exp+defi-wallet"/>
> 
>       </intent-filter>
> 
>      <intent-filter android:autoVerify="true" data-generated="true">
> 
>         <action android:name="android.intent.action.VIEW"/>
> 
>         <data android:scheme="https" android:host="1b1.art" android:pathPrefix="/claim"/>
> 
>         <category android:name="android.intent.category.BROWSABLE"/>
> 
>         <category android:name="android.intent.category.DEFAULT"/>
> 
>       </intent-filter>
> 
>       <intent-filter data-generated="true">
> 
>         <action android:name="android.intent.action.VIEW"/>
> 
>          <data android:scheme="dev.ooo.opn.wallet"/>
> 
>         <data android:scheme="exp+defi-wallet"/>
> 
>         <category android:name="android.intent.category.BROWSABLE"/>
> 
>         <category android:name="android.intent.category.DEFAULT"/>
> 
>       </intent-filter>
> 
> ```
> 
> 
> 
> Callback url: `dev.ooo.opn.wallet://{MY_DOMAIN}/android/dev.ooo.opn.wallet/callback`
> 
> 
> 
> Actual:
> 
> 
> 
> https://user-images.githubusercontent.com/32274987/207081814-d9d635f8-ce7d-4cf8-8585-79d479d2be98.png
> 
> 
> 
> 
> 
> Expected:
> 
> 
> 
>  To have only one or automatically open the app both works as expected
> 
> 
> 
> Env:
> 
> 
> 
> ```
> 
> "react-native-auth0": "2.15.0",
> 
> "expo": "~46.0.17",
> 
> "react-native": "0.69.6"
> 
> ```

Hey man, how did you solve it?

**Comment by cowlabs-xyz** [comment]
If anyone still experiences this the reason could be that you are not using a custom scheme with your react native auth0.

The issue for me was:
- I was not using a customScheme for my Auth0 callback url. So Auth0 would default to use my apps package name as the scheme
- Since expo automatically adds the apps package name as a scheme on the apps MainActivity it would mean that Auth0s RedirectActivity would be a duplicate of the scheme. The package name scheme would be present in both the auth0 RedirectActivity and MainActivity leading to two options for handling the same scheme

So the easy fix a was to use customScheme.

From docs it does say it is mandatory on react native to use custom scheme, however I ignored reading the docs before starting...

app.json or app.config.js
`{
  "expo": {
    ...
    "plugins": [
      [
        "react-native-auth0",
        {
          "domain": "YOUR_AUTH0_DOMAIN",
          "customScheme": "YOUR_CUSTOM_SCHEME"
        }
      ]
    ]
  }
}`

remember to pass the customScheme as parameter in the 
`Auth0.authorize( {customScheme: "YOUR_CUSTOM_SCHEME"} )`
and same in 
`clearSession({customScheme: "YOUR_CUSTOM_SCHEME"})`

If you are using TypeScript make sure to update your types to latest version, as older versions do not have correct function signature for clearSession()

`"@types/react-native-auth0": "^2.17.4"`
